Akuma Tags Fix

KenjieDec opened the pull request:
- Removed "Pages" filter — It doesn't work
- Added "Other Tags" filter
- Added "Image Set" category filter
- Added Filter limit warning
- Fixed problem on tags with spaces <sub>( For example: `big penis`. Before:`Male tag contains big. Title contains penis`. After:`Male tag contains big penis` )<sub/>

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

KenjieDec committed:
Fix

- Removed "Pages" filter — They don't work
- Added "Other Tags" filter
- Added Filter limit warning
- Fixed problem on tags with spaces ( e.g. bxg brxxsts )

KenjieDec committed:
Extension version: 2 -> 3

KenjieDec committed:
typo fix

AwkwardPeak7 commented on src/all/akuma/build.gradle:
make sure your line endings are `lf`

AwkwardPeak7 commented on src/all/akuma/src/eu/kanade/tachiyomi/extension/all/akuma/Akuma.kt:
while you are at it, remove the preference for iconfiied tags, and make it default.

I belive having a preference for such a thing isn't necessary

KenjieDec replied:
Ok 👍

KenjieDec committed:
Change line endings, make prefrence default

- Line endings: CRLF -> LF
- Iconified Tag Preference -> Iconified Tag is default

KenjieDec committed:
Missing space

AwkwardPeak7 commented on src/all/akuma/build.gradle:
this file still have different line ending as evident by github showing whole file having difference

AwkwardPeak7 commented on src/all/akuma/src/eu/kanade/tachiyomi/extension/all/akuma/Akuma.kt:
```diff
@@ -176,39 +166,39 @@ class Akuma(
+        if (lang != "all") {
+            finalQuery.add("language: $akumaLang$")
+        }
+        filters.forEach { fIt ->
```
```suggestion
        filters.forEach { filter->
```
since fIt is ambigious imo

KenjieDec committed:
Fix1

KenjieDec committed:
Change

KenjieDec committed:
Change2

AwkwardPeak7 commented on src/all/akuma/src/eu/kanade/tachiyomi/extension/all/akuma/Akuma.kt:
```diff
@@ -268,6 +263,9 @@ class Akuma(
                     name = "Chapter"
+                    date_upload = SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.ENGLISH).apply {
+                        timeZone = TimeZone.getTimeZone("UTC")
+                    }.parse(client.newCall(chapterListRequest(manga)).execute().asJsoup().select(".date .value>time").text())?.time!!
```
just remove override for `fetchChapterList` and provide implementation in `chapterListParse` where you will get the response directly

AwkwardPeak7 commented on src/all/akuma/src/eu/kanade/tachiyomi/extension/all/akuma/Akuma.kt:
```diff
@@ -176,39 +169,39 @@ class Akuma(
+                is TextFilter -> {
+                    if (filter.state.isNotEmpty()) {
+                        finalQuery.addAll(
+                            filter.state.split(",").map {
```
filter out empty ones, for example user can put 2 commas torgether

KenjieDec committed:
Change3

- Removed override for fetchChapterList and provide implementation in chapterListParse
- Filter out empty tags ( For example: user can put 2 commas together )

AwkwardPeak7 commented on src/all/akuma/src/eu/kanade/tachiyomi/extension/all/akuma/Akuma.kt:
```diff
@@ -248,28 +243,29 @@ class Akuma(
+                name = "Chapter"
+                date_upload = SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.ENGLISH).apply {
+                    timeZone = TimeZone.getTimeZone("UTC")
+                }.parse(response.asJsoup().select(".date .value>time").text())?.time!!
```
move the date format to class val so it doesn't get initialized every time

KenjieDec committed:
Change4

Moved Date Format to Class val

AwkwardPeak7 committed:
try catch for date parse

AwkwardPeak7 committed:
Update Akuma.kt
